### PR TITLE
feat: create correct notification content for every event type-  SQSERVICES-1700

### PR DESCRIPTION
**Pull request by MarcinR**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1700" title="SQSERVICES-1700" target="_blank">SubtaskSQSERVICES-1700</a>  Refactor NotificationService for possible reuse
  </td></table>
  <br />
 

This pull requests implements Phase 4 from the [new notification extension plan](https://wearezeta.atlassian.net/wiki/spaces/IOS/pages/647725130/Notification+Service+Extension+2.0).
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability

### CODE REVIEW

**Review by marcoconti83 — CHANGES_REQUESTED** [review]

**Review by marcoconti83 — CHANGES_REQUESTED** [review]
Overall looks good, but I left some comments. 

In order to get the big picture, could you please edit the PR description to specify what is the purpose of this PR? Same like we did for https://github.com/wireapp/wire-ios/pull/5888. 
I.e., Is this PR implementing Step 4 entirely? Is it only a part of it, and more work need to follow? If so, what part is this PR addressing?

File: Wire Notification Service Extension/Simple/Helpers/EventMessageExtractor.swift [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-//
-// You should have received a copy of the GNU General Public License
-// along with this program. If not, see http://www.gnu.org/licenses/.
-//
-
-import Foundation
-import WireDataModel
-
-class EventMessageExtractor {
-    func extractMessage(fromDecodedEvent event: ZMUpdateEvent) throws -> String {
```
[the remark this answers is not in the document]
Comment (MarcinR) [reply Wire Notification Service Extension/Simple/Helpers/EventMessageExtractor.swift:23]: I removed EventMessageExtractor and create NotificationContentProvider. GenericMessage is created in ZMLocalNotification class

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -10146,6 +10158,7 @@
 					"@executable_path/Frameworks",
 				);
 				ONLY_ACTIVE_ARCH = YES;
+				PROVISIONING_PROFILE_SPECIFIER = "Wire iOS Development Debug";
```
[the remark this answers is not in the document]
Comment (MarcinR) [reply Wire-iOS.xcodeproj/project.pbxproj]: yes, that shouldn't be here. I removed it in step3 PR, but it wasn't mergered to this branch yet

File: Wire Notification Service Extension/Simple/Helpers/EventMessageExtractor.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcinR) [reply Wire Notification Service Extension/Simple/Helpers/EventMessageExtractor.swift:23]: also this whole step4

**Pull request merged**